crates/uv-distribution-filename/src/wheel.rs

```diff
@@ -85,45 +97,47 @@ impl WheelFilename {
             "{}-{}-{}",
             self.name.as_dist_info_name(),
             self.version,
-            self.get_tag()
+            self.tags
         )
     }
 
-    /// Parse a wheel filename from the stem (e.g., `foo-1.2.3-py3-none-any`).
-    pub fn from_stem(stem: &str) -> Result<Self, WheelFilenameError> {
-        Self::parse(stem, stem)
+    /// Return the wheel's Python tags.
+    pub fn python_tags(&self) -> &[LanguageTag] {
+        match &self.tags {
+            WheelTag::Small { small } => std::slice::from_ref(&small.python_tag),
+            WheelTag::Large { large } => large.python_tag.as_slice(),
+        }
     }
 
-    /// Get the tag for this wheel.
-    fn get_tag(&self) -> String {
-        if let ([python_tag], [abi_tag], [platform_tag]) = (
-            self.python_tag.as_slice(),
-            self.abi_tag.as_slice(),
-            self.platform_tag.as_slice(),
-        ) {
-            format!("{python_tag}-{abi_tag}-{platform_tag}",)
-        } else {
-            format!(
-                "{}-{}-{}",
-                self.python_tag
-                    .iter()
-                    .map(ToString::to_string)
-                    .collect::<Vec<_>>()
-                    .join("."),
-                self.abi_tag
-                    .iter()
-                    .map(ToString::to_string)
-                    .collect::<Vec<_>>()
-                    .join("."),
-                self.platform_tag
-                    .iter()
-                    .map(ToString::to_string)
-                    .collect::<Vec<_>>()
-                    .join("."),
-            )
+    /// Return the wheel's ABI tags.
+    pub fn abi_tags(&self) -> &[AbiTag] {
+        match &self.tags {
+            WheelTag::Small { small } => std::slice::from_ref(&small.abi_tag),
```

Comment:
I love this function. One of my favorites in all of `std`.

Reply:
So so good.